### ISSUE TITLE
Error creating Enterprise Application with Maven

### ISSUE THREAD

**Issue by criscubillosedi** [apache/netbeans#3552]
### Apache NetBeans version

Latest release

### What happened

When I try to create an Enterprise application with Maven the process fails and generates a corrupt project. The last version of Netbeans where it worked correctly for me is 8.2, which I still have to generate the base projects. All new versions of netbeans fail the same way.

### How to reproduce

1) New Project
2) Categories: Java with Maven
3) Enterprise application
4) fill in the fields normally
5) Select any server
6) Java EE version: Jakarta EE 8
7) Check create EJB Module
8) Check Web App module and assembly module
9) Finish

<img width="714" alt="wizard01" src="https://user-images.githubusercontent.com/67790211/152344978-c66402a7-b86c-4462-b188-6811efba00b3.png">

<img width="904" alt="wizard02" src="https://user-images.githubusercontent.com/67790211/152344998-c6426f20-e720-4acd-bcc1-46b7118dfd20.png">

**Project creation result:**

Log output:

Generating project in Batch mode
Archetype not found in any catalog. Falling back to central repository.
Add a repository with id 'archetype' in your settings.xml if archetype's repository is elsewhere.
----------------------------------------------------------------------------
Using following parameters for creating project from Archetype: webapp-jakartaee8:1.0.0
----------------------------------------------------------------------------
Parameter: groupId, Value: testapp
Parameter: artifactId, Value: enterpriseApplicationWithMaven-ejb
Parameter: version, Value: 1.0.0
Parameter: package, Value: testapp.enterpriseApplicationWithMaven-ejb
Parameter: packageInPathFormat, Value: testapp/enterpriseApplicationWithMaven-ejb
Parameter: package, Value: testapp.enterpriseApplicationWithMaven-ejb
Parameter: groupId, Value: testapp
Parameter: artifactId, Value: enterpriseApplicationWithMaven-ejb
Parameter: version, Value: 1.0.0
------------------------------------------------------------------------
BUILD FAILURE
------------------------------------------------------------------------
Total time:  4.155 s
Finished at: 2022-02-03T09:34:46-03:00
------------------------------------------------------------------------
Failed to execute goal org.apache.maven.plugins:maven-archetype-plugin:3.1.2:generate (default-cli) on project enterpriseApplicationWithMaven: Unable to add module to the current project as it is not of packaging type 'pom' -> [Help 1]

Full log is attached:
[project_creation.log](https://github.com/apache/netbeans/files/7994587/project_creation.log)

**Finally:**
A series of directories are created with the correct structure, but the pom.xml is not created properly and all modules seem to be of WAR type.

<img width="311" alt="project-view" src="https://user-images.githubusercontent.com/67790211/152346244-8cc0770e-300f-4354-8106-f5a0a3a83c34.png">

### Did this work correctly in an earlier version?

- [X] This used to work!

### Operating System

macOS Monterey, Windows 10, Windows 11

### JDK

zulu11, zulu8, jdk1.7.0_80, jdk1.8.0_271

### Apache NetBeans packaging

Apache NetBeans provided installer

### Anything else

I assume that the problem lies in the archetype used

### Are you willing to submit a pull request?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow the Apache Software Foundation's [Code of Conduct](https://www.apache.org/foundation/policies/conduct.html)

**Comment by emaphis** [comment]
Can confirm on Windows 10, JDK 17, NetBeans 12.6
NetBeans is creating the wrong structure.  It supposed to create a parent POM app with several children modules.  
But it creates nested apps with no parent relationship.

**Comment by matthiasblaesing** [comment]
@juneau001 could you please have a look at this? If I see this correctly, JEE 7 is still implemented correctly, all the possible artifacts for JEE got their own archetypes:

- EAR: [org.codehaus.mojo.archetypes:ear-javaee7:1.0](https://search.maven.org/artifact/org.codehaus.mojo.archetypes/ear-javaee7/1.0/maven-archetype)
- EJB-JAR: [org.codehaus.mojo.archetypes:ejb-javaee7:1.1](https://search.maven.org/artifact/org.codehaus.mojo.archetypes/ejb-javaee7/1.1/maven-archetype)
- WAR: [org.codehaus.mojo.archetypes:webapp-javaee7:1.1](https://search.maven.org/artifact/org.codehaus.mojo.archetypes/webapp-javaee7/1.1/maven-archetype)
- Appclient: [org.codehaus.mojo.archetypes:appclient-javaee7:1.1](https://search.maven.org/artifact/org.codehaus.mojo.archetypes/appclient-javaee7/1.1/maven-archetype)
- Bundling POM Projekt: [org.codehaus.mojo.archetypes:pom-root:1.1](https://search.maven.org/artifact/org.codehaus.mojo.archetypes/pom-root/1.1/maven-archetype)

If I remember correctly, the EAR bundles the EJB-JAR and WAR into a single artifact and combines these into the server side artifact. The applient artifact is used by clients to access the server side resources and the POM projects wraps all these artifacts together into a manageable bundle.

With recent JEE versions the old rigid style vanished and you could bundle EJBs and Webparts into a single artifact (either an EJB jar or a WAR file ) and deploy these.

Looking at the definitions for JEE 8 + 9 I see, that all artifacts use the same archetypes and I don't see how they could generate all the artifacts described above.

TL;DR: I think the support for JEE 8 + 9 only covers only the Web Tier and the definitions for the FullProfile need to be redone.

**Comment by juneau001** [comment]
I will look into this project type.  However, the trend since Java EE 7 is to use WAR files and JAR files for application deployment, rather than EAR files for deployment, since they are lighter weight.  EAR files were mainly used in Java EE 7 or older.  It seems like this functionality still works if you choose Java EE 7 or below, as you had mentioned.

Do we want to make this functionality work within Apache NetBeans for Java EE 8 and Jakarta EE projects even though it is no longer a standard project format for these newer versions?

**Comment by asbachb** [comment]
Afaik JavaEE/JakartaEE spec require backwards compatibility so the support won't go away soon.

But it seems a legitimate question if these types should be proposed by NetBeans for new projects. Afaik you still need an EAR when you bundle a resource adapter.

**Comment by OndroMih** [comment]
I hit this issue too. The reason why the project ends in a corrupted stat is that for newer Java EE and Jakarta EE versions, it uses the same archetype for all generated Maven modules. And the archteype generates a WAR module. Therefore we'll get a WAR module, with multiple nested WAR modules. This is not only wrong but invalid, because the top-level module isn't POM but WAR.

If we decide that Netbeans won't support EAR project generation for newer Jakarta EE versions, we should then remove the archetype for the newer versions of Jakarta EE in https://github.com/apache/netbeans/blob/f0ed696648470561d9e05c03db965812893b3f51/enterprise/maven.j2ee/src/org/netbeans/modules/maven/j2ee/ui/wizard/archetype/J2eeArchetypeFactory.java#L118, so that users can choose the newer versions and get a corrupted project.

It should be reasonably easy to add support for EAR though, by adding appropriate archetypes, because they are very simple. The source code repository for old archetypes at codehaus.org is gone, however, their source code is still available in Maven Central and we can loosely base the new archtypes on that (but not simply fork it, because it's not released under any opensource license AFAIK). I can have a look into this. @juneau001, would it be OK to contribute to your repository for the WEB archetype? Or better to do it in my own repo/fork and release the new archetypes to Maven Central by myself?

**Comment by melanysoft** [comment]
Is there any resolution to this issue ?
It affects every Netbeans version from 14-16, so the only way to build EAR Apps is by using Ant (or another IDE).
I tried to create an Enterprise Application without WAR module but same error occurred.

**Comment by OndroMih** [comment]
There's currently no solution I'm aware of. As I wrote, it should be fairly easy to support EARs but somebody has to contribute the solution. I'd like to do it but not enough free time for that in the coming weeks :( Send me a message on Twitter or email and maybe I'll be able to help you.
